cg/meta/workflow/mutant/quality_controller/utils.py

```diff
@@ -0,0 +1,106 @@
+from cg.apps.lims.api import LimsAPI
+from cg.constants.constants import MutantQC
+from cg.constants.lims import LimsArtifactTypes, LimsProcess
+from cg.exc import CgError, LimsDataError
+from cg.meta.workflow.mutant.metrics_parser.metrics_parser import MetricsParser
+from cg.meta.workflow.mutant.metrics_parser.models import SampleResults
+from cg.meta.workflow.mutant.quality_controller.models import (
+    MutantPoolSamples,
+    QualityMetrics,
+)
+from genologics.entities import Artifact
+from genologics.entities import Sample as LimsSample
+from cg.models.cg_config import LOG
+from cg.services.sequencing_qc_service.quality_checks.utils import sample_has_enough_reads
+from cg.store.models import Case, Sample
+
+from pathlib import Path
+
+from cg.store.store import Store
+
+
+def has_valid_total_reads(
+    sample: Sample,
+    external_negative_control: bool = False,
+    internal_negative_control: bool = False,
+) -> bool:
+    if external_negative_control:
+        return external_negative_control_sample_has_enough_reads(reads=sample.reads)
+
+    if internal_negative_control:
+        return internal_negative_control_sample_has_enough_reads(reads=sample.reads)
+
+    return sample_has_enough_reads(sample=sample)
+
+
+def external_negative_control_sample_has_enough_reads(reads: int) -> bool:
+    return reads < MutantQC.EXTERNAL_NEGATIVE_CONTROL_READS_THRESHOLD
+
+
+def internal_negative_control_sample_has_enough_reads(reads: int) -> bool:
+    return reads < MutantQC.INTERNAL_NEGATIVE_CONTROL_READS_THRESHOLD
+
+
+def get_internal_negative_control_id_for_case(lims: LimsAPI, case: Case) -> str:
+    """Query lims to retrive internal_negative_control_id for a mutant case sequenced in one pool."""
+
+    sample_internal_id = case.sample_ids[0]
+    internal_negative_control_id: str = lims.get_internal_negative_control_id_from_sample_in_pool(
+        sample_internal_id=sample_internal_id, pooling_step=LimsProcess.COVID_POOLING_STEP
+    )
+    return internal_negative_control_id
+
+
+def get_internal_negative_control_sample_for_case(
+    case: Case,
+    status_db: Store,
+    lims: LimsAPI,
```

Comment:
Passing statusdb and lims as arguments seems redundant to me, would it make sense to include these methods as part of the mutant analysis api?

Reply:
Same for the rest of the methods below Could they be moved either to the analysis API or the quality controller?

Reply:
Agreed! Moved it to the quality_controller!

Reply:
I will move them to the quality_controller then!